### PR TITLE
Icon names, and symbolic.  Makes Adwaita 43 happy.

### PR DESCRIPTION
**Pull request by b4n**
See #3613.
Some of the changes here make sense, some don't really but are still needed not to look terrible with Adwaita 43.  See each commit for details.

@techee @eht16 that's kind of how it looks to dodge the Adwaita 43 issues I could see.

### CODE REVIEW

**Comment by elextr** [comment]
This needs to be tested on a wide range of distros, not just Debian, and not just Gnome, Cinnamon, XFCE, etc and Windows.

**Comment by techee** [comment]
> This needs to be tested on a wide range of distros, not just Debian, and not just Gnome, Cinnamon, XFCE, etc and Windows.

I don't think it needs to be tested on all thinkable distributions (just tried briefly on macOS and it seems to work), it should work the same way everywhere.

On the other hand it's not just a few functions renaming and a few strings renaming as I thought it would be and we are just too close to the release (and too close to the midnight - at least those of us living in the correct hemisphere and time zone ;-) so I'd say better to postpone this after 2.0.

**Comment by elextr** [comment]
> on all thinkable distributions

No, just Ubuntu, Mint, Arch probably.  The point is that the change is relying on icon themes, and distros are the thing that provides them, and possibly differing sets on each desktop, as far as we can tell all but Debian Adwaita provide sensible `gtk-clear` icons (since nobody has complained), but do they provide sensible `edit-clear-symbolic` icons?

Without checking we just don't know.

> postpone this after 2.0.

Agreed, until we are confident we won't break a whole lot of other systems to fix one.

> too close to the midnight

Go to bed while I have breakfast :-)

**Comment by b4n** [comment]
Agreed, it grew way too out of hand to be merged less than 24 hours before the deadline.
And I agree with @elextr that it likely requires more testing than just hoping every other distro is shipping a well-behaving theme (better behaving than Debian's Adwaita 43) by default, although @techee has a point that as the icons that please Adwaita are used by GTK itself, it's *likely* not to cause much trouble.

Anyway, on top of this some of the changes here I'm happy with, some I am not so much (as mentioned in the commit messages), so it's not even like it's obviously the right thing to do down the road.

Let's just cross fingers and hope Debian does something about it, and nobody else is shipping a broken Adwaita icon theme… or that none of our user actually use the broken ones. :crossed_fingers:

**Comment by elextr** [comment]
> Let's just cross fingers and hope Debian does something about it, and nobody else is shipping a broken Adwaita icon theme… or that none of our user actually use the broken ones.

No chance unless Murphy goes on holidays :smiling_imp:

**Comment by techee** [comment]
> although @techee has a point that as the icons that please Adwaita are used by GTK itself, it's likely not to cause much trouble.

I was looking for this page (which I couldn't find yesterday) for stock icons and their replacements:

https://developer-old.gnome.org/gtk3/stable/gtk3-Stock-Items.html#GTK-STOCK-ABOUT:CAPS

This has been advised by GTK since 3.10 and many applications started using these icons in the meantime so I think we are pretty safe regarding the icons themselves. IMO it's mostly just a matter of using the correct GTK API (plus deprecating some plugin API like `ui_image_menu_item_new()` and having a substitute for it).

**Comment by cwendling** [comment]
@techee yeah, but here just using the icon name won't solve it, you also need to use the symbolic version for the problematic icons.

**Comment by techee** [comment]
> @techee yeah, but here just using the icon name won't solve it, you also need to use the symbolic version for the problematic icons.

Ah, OK, I missed that. But then it's really a problem of the icon theme I think.

**Comment by polter-rnd** [comment]
I've tried this patch on Fedora 38 with Adwaita-dark theme, and all icons are dark on dark: 
Screenshot from 2023-10-20 19-52-19

**Comment by b4n** [comment]
I just rebased and updated to try and install a fallback icon instead of the questionable change.  I also dropped the hack not to introduce a new string, which is irrelevant now.
With the last change, we could get rid of most of the other changes, but in some instances using the symbolic icon seems reasonable, so we could do it anyway.  But we could also be conservative and just include the missing icons and call it a day.

Opinions?

**Comment by techee** [comment]
Isn't it this problem: https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/288 ?

The way I understand it is that the fix was to create `adwaita-icon-theme-legacy` which contains the non-symbolic variants from which `adwaita-icon-theme` inherits. Distros should then ship both of these themes (this by the way happens in msys2 too when you look at the dependencies of Geany).

So do we have to do anything about it at all?

**Comment by b4n** [comment]
Hum yeah but IIUC it is *another* theme with the Legacy suffix, e.g. not gonna fix a regular install unless it defaults to that, does it?

**Comment by techee** [comment]
It will fix it as long as the `-legacy` theme is installed together with `adwaita-icon-theme`. Unfortunately it isn't on bookworkm:

https://packages.debian.org/bookworm/adwaita-icon-theme

but it is on trixie (and probably all subsequent Debian releases):

https://packages.debian.org/trixie/adwaita-icon-theme

`adwaita-icon-theme` itself inherits from the `-legacy` theme

https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/blob/master/index.theme?ref_type=heads#L5

so when it's installed, it will pull all missing icons from it.

So to sum up, this problem will probably exist on bookworm and derived distros but rather than doing some crazy things in Geany, I'd suggest that we just instruct affected users to install the `-legacy` theme and update `index.theme` of `adwaita-icon-theme` to inherit from it. Any future Debian releases should have this fixed.

**Comment by techee** [comment]
I've just noticed that in trixie the `-legacy` package is just a suggested dependency so it will probably not be installed by default, will it?
